pkg/virt-launcher/virtwrap/agent-poller/agent_poller_test.go

```diff
@@ -20,413 +20,140 @@
-					Domain:    "bobs",
-					LoginTime: 99999,
-				},
-			}
-
-			Expect(err).ToNot(HaveOccurred(), "users should be parsed normally")
-			Expect(users).To(Equal(expectedUsers))
+			Expect(commandExecutions).To(Equal(expectedExecutions))
 		})
 
-		Context("with AsyncAgentStore", func() {
-
-			It("should store and load the data", func() {
-				var agentStore = NewAsyncAgentStore()
-				agentVersion := "4.1"
-				agentStore.Store(GET_AGENT, agentVersion)
-				agent := agentStore.GetGA()
+		It("executes the agent commands based on the time interval specified", func() {
+			const interval = 1
+			const expectedExecutions = 3
 
-				Expect(agent).To(Equal(agentVersion))
-			})
+			commandExecutions := runPollAndCountCommandExecution(interval, expectedExecutions, 0)
 
-			It("should fire an event for new sysinfo data", func() {
-				var agentStore = NewAsyncAgentStore()
-
-				fakeInfo := api.GuestOSInfo{
-					Name:          "TestGuestOSName",
-					KernelRelease: "1.1.0-Generic",
-					Version:       "1.0.0",
-					PrettyName:    "TestGuestOSName 1.0.0",
-					VersionId:     "1.0.0",
-					KernelVersion: "1.1.0",
-					Machine:       "x86_64",
-					Id:            "testguestos",
-				}
-				agentStore.Store(GET_OSINFO, fakeInfo)
-
-				Expect(agentStore.AgentUpdated).To(Receive(Equal(AgentUpdatedEvent{
-					Type:       GET_OSINFO,
-					DomainInfo: api.DomainGuestInfo{OSInfo: &fakeInfo},
-				})))
-			})
+			Expect(commandExecutions).To(Equal(expectedExecutions))
+		})
 
-			It("should not fire an event for the same sysinfo data", func() {
-				var agentStore = NewAsyncAgentStore()
-				fakeInfo := api.GuestOSInfo{
-					Name:          "TestGuestOSName",
-					KernelRelease: "1.1.0-Generic",
-					Version:       "1.0.0",
-					PrettyName:    "TestGuestOSName 1.0.0",
-					VersionId:     "1.0.0",
-					KernelVersion: "1.1.0",
-					Machine:       "x86_64",
-					Id:            "testguestos",
-				}
+		It("executes the agent commands in increasing intervals at initial run", func() {
+			const interval = 30
+			const expectedExecutions = 2
 
-				agentStore.Store(GET_OSINFO, fakeInfo)
-				Expect(agentStore.AgentUpdated).To(Receive(Equal(AgentUpdatedEvent{
-					Type:       GET_OSINFO,
-					DomainInfo: api.DomainGuestInfo{OSInfo: &fakeInfo},
-				})))
+			t := pollInitialInterval + timeoutBuffer
+			commandExecutions := runPollAndCountCommandExecution(interval, expectedExecutions, t)
 
-				agentStore.Store(GET_OSINFO, fakeInfo)
-				Expect(agentStore.AgentUpdated).ToNot(Receive())
-			})
+			Expect(commandExecutions).To(Equal(expectedExecutions))
 		})
 	})
 })
+
+// countSignals counts the number of signals received through the `done` channel
+// The timeout is calculated from the first received signal.
+func countSignals(done <-chan struct{}, timeout time.Duration) int {
+	var counter int
+	t := time.After(timeout)
+	for {
+		select {
+		case <-t:
+			return counter
+		case <-done:
+			if counter == 0 {
+				t = time.After(timeout)
```

Comment:
can you please `t.Stop` ?

Reply:
You cannot stop a channel.

Reply:
timer, why channel ?

Reply:
You right,
consider
``` 
//... If efficiency is a concern, use NewTimer
// instead and call Timer.Stop if the timer is no longer needed.
```
And then you can

Reply:
So you want me to quote the `time.After` documentation in the code base to make it clearer for you that this is not a timer but a channel?

Reply:
no, i just suggest to change the name of it to `c` or to use timer :)